core/presentation/designsystem/src/main/java/com/nolawiworkineh/designsystem/components/TaskyTextField.kt

```diff
@@ -58,15 +57,16 @@ fun TaskyTextField(
     modifier: Modifier = Modifier,
     keyboardType: KeyboardType = KeyboardType.Text,
     isError: Boolean = false,
-    errorMessage: String? = null,
     imeAction: ImeAction = ImeAction.Done,
 ) {
     var isFocused by remember { mutableStateOf(false) }
+    var hasBeenFocused by remember { mutableStateOf(false) }
+    var showErrorIndicator by remember { mutableStateOf(false) }
```

Comment:
If these remember states are just for temporary testing, please ignore this comment :D 

I'd strongly recommend to put all UI states in a ViewModel, since here they won't survive config changes. After a screen rotation or dark/light theme switch, all your `hasBeenFocused` and `showErrorindicator` states will be reset. With `rememberSaveable` you could work around that, but then your UI will have too many responsibilities, since it also needs to decide how the state is updated.

With a class like this, you can encapsulate your textfields’ error states:

```
data class LoginUiState(
  val email: TextFieldState = TextFieldState(),
  val password: PasswordState = PasswordState(),
  val isLoading: Boolean = false,
  val isLoggedIn: Boolean = false,
  val errorMessage: UiText? = null // Here you can have multiple properties, one for each textfield if needed
)
```